### DISCUSSION TITLE
Sonoff POW R2 - DO NOT CONNECT ANY SWITCH OR SENSOR

### DISCUSSION THREAD

**Discussion by DonRichie** [arendst/Tasmota#11313] [Q&A] [answered]
**Hello,**

I would like to reopen Issue #4732 and feed it with additional information which might change the developers opinion.

The discussion there was that it might be harmful for humans to use GPIO4 and GPIO5
But I believe there is a misunderstanding, since the pcb layout differs between the first "Sonoff POW" and the "Sonoff POW R2".

I believe in the other issue the responding people thought that the requester intended to do some soldering next to the AC connection, which would have been indeed a bad idea. But this is not the case.

### Sonoff POW R2 Layout
The tasmota documentation for the "Sonoff POW R2" shows an image of the "Sonoff POW" in its first version, but not the "Sonoff POW R2":
https://tasmota.github.io/docs/devices/Sonoff-Pow-R2/

This is the upper side of a "Sonoff POW R2":
<img src="https://s.ricl.de/sonoff_pow_r2/upper_side.jpg" width="300px">

The difference is that the "POW R2" has two additional pins to solder a pin header to. Those are GPIO4 and GPIO5.
There is also by design a hole which can be used to put cables through it.

Please see this picuture how I used the pin headers and the hole to connect some Dupont cables:
<img src="https://s.ricl.de/sonoff_pow_r2/case1.jpg" width="300px">
I am at any time miles away from the AC part on the PCB. (The hole in the case was also already there).

### My interpretation
I believe the manufacturer on purpose made the two GPIO Pins available for hobbyists and intended that we might use them somehow.

In fact there is an other model named "Sonoff TH10" which is using the same case and a smiliar PCB layout.
The TH10 is made for attaching a sensor utilizing the hole in the case, but doesn't provide the two additional GPIO pins.
See: [Sonoff TH10 (from google)](http://ewelink.coolkit.cc/wp-content/uploads/2016/09/Sonoff_TH10_TH16_wiring_diagram-1.jpg)

### Feature Request
Please make GPIO4 and GPIO5 usable for arbitrary hardware when selecting "Sonoff POW R2" as model in tasmota.

(Also please update the documentation with a picture of the correct model)

### All Photos (Feel free to use my pictures for anything)
case: <img src="https://s.ricl.de/sonoff_pow_r2/case1.jpg" width="50px"><img src="https://s.ricl.de/sonoff_pow_r2/case2.jpg" width="50px"><img src="https://s.ricl.de/sonoff_pow_r2/case3.jpg" width="50px"><img src="https://s.ricl.de/sonoff_pow_r2/case4.jpg" width="50px">
lower_side: <img src="https://s.ricl.de/sonoff_pow_r2/lower_side.jpg" width="50px"><img src="https://s.ricl.de/sonoff_pow_r2/lower_side2.jpg" width="50px"><img src="https://s.ricl.de/sonoff_pow_r2/lower_side3.jpg" width="50px">
lower_side_ic_zoom: <img src="https://s.ricl.de/sonoff_pow_r2/lower_side_ic_zoom1.jpg" width="50px"><img src="https://s.ricl.de/sonoff_pow_r2/lower_side_ic_zoom2.jpg" width="50px"><img src="https://s.ricl.de/sonoff_pow_r2/lower_side_ic_zoom3.jpg" width="50px">

[lower_side_highresolution1](https://s.ricl.de/sonoff_pow_r2/lower_side_highresolution1.jpg) [lower_side_highresolution2](https://s.ricl.de/sonoff_pow_r2/lower_side_highresolution2.jpg) [lower_side_highresolution1](https://s.ricl.de/sonoff_pow_r2/lower_side_highresolution3.jpg)

**Comment by sfromis** [comment]
**STOP** Dangerous.

Do not attach anything whatsoever to the Sonoff Pow R2 board. All of the board can be at mains power level. The hole in the case is lazy manufacturing, using the same plastic piece as used on the Sonoff TH devices without this danger.

The old warning still applies to the R2. https://templates.blakadder.com/sonoff_Pow_R2.html

**Reply by DonRichie** [reply]
Can you please eloberate this?

I don't see why plugging in a USB power adapter for a nodemcu would be different from using an onboard voltage regulator.
As I mentioned they are providing the model TH10 which has a relay and at the same time a sensor.

**Reply by sfromis** [reply]
The reason for energy monitoring devices being very different from a USB power adapter is that that the circuit needed for energy monitoring needs mains voltage. The TH10/TH16 does not have this energy monitoring circuit, and therefore a different design.

Just tested (with great care) a Sonoff Pow R2, hooked up to mains. Of course, my multimeter showed 230V potential between live and neutral input. Same (!) potential between the supposed low voltage ground and neutral input. 3.3v higher potential (233.3v) between supposed low voltage vcc and neutral input. Confirmed 3.3v potential difference between 3.3v and gnd on the supposed low voltage side. The whole low voltage side is at mains level, existing on the 3.3v difference around 230v. 

You should never hook things up to 230v, unless wires and protection are dimensioned for mains (16A). The whole board checked out to be at mains level. I consider it a **serious** design flaw to have those tempting IO4/IO5 connections.

**Reply by timaseth** [reply]
On similar lines; `Best to leave all free GPIOs as they are on _ALL_ things PM/EM. Not worth the risk.`

**Reply by DonRichie** [reply]
Thank you sfromis, your explanation helps me a lot.
I understood there are additional potentials >=230V between VCC/GND and Neutral/Live which are the problem. And they are issued by the special kind of PCB layout necessary for the power monitoring chip.

Can you answer me a final question: In a PCB which doesn't have a power monitoring chip.
What potentials would be expected between:
VCC and Neutral
VCC and Live
GND and Neutral
GND and Live

It should be something harmless then, right?

**Reply by ascillato** [reply]
In devices like a nodemcu or a sonoff basic, there is no electrical ground reference between Mains AC and 3.3v/GND rail. So, you will measure OPEN/HIGH IMPEDANCE with a DMM.

Please, use a nodemcu for sensors. That device is safer. 

NEVER MODIFY A POWER MEASUREMENT DEVICE.

**Reply by DonRichie** [reply]
@ascillato: Okay, this means my multimeter shows 0V in all mentioned cases, right?

Edit: your edit explained it, High Impedance, I understood. Thank you

**Comment by ascillato** [comment, marked as answer]
This is so DANGEROUS. I'm sorry but it is very obvious that you didn't read any of the warnings and explanations in old issues and in the docs!

All power measurements devices are grounded to mains AC. The pins of the 3.3V is grounded to mains!

Please search in old issues and you will see the experiments people have done and how they burnt devices and also received electric shocks.

### You can burn your house and or kill any member of your family. Please DO NOT MODIFY POWER MEASUREMENT DEVICES.

Your request is not going to be added. Very DANGEROUS.

**Comment by DonRichie** [comment]
Thank you all for your answers and helping me no to die.
I won't attach a sensor to the mentioned device.

Please feel free to use my pictures for the tasmota documentation.

**Comment by sfromis** [comment]
When I upgraded my first Sonoff Pow R2 to Tasmota, I also soldered headers to those "extra" pins, just in case. When I learned about the  dangers, I snipped off those headers again. I'd recommend you do the same, as a reminder :-)
